### PR TITLE
Deal with scipy unceremoniously removing `wrap_function`

### PR DESCRIPTION
**Pull request by mpharrigan**

### CODE REVIEW

**Comment by mpharrigan** [comment]
This fixes the issue seen in https://github.com/quantumlib/ReCirq/pull/183 and https://github.com/quantumlib/ReCirq/pull/183 fixes the issue seen in this

**Comment by madcpf** [comment]
Thanks for fixing this!

**Review by balopat — APPROVED** [review]
LGTM  

For the record, it is a verbatim copy from https://github.com/scipy/scipy/blob/maintenance/1.6.x/scipy/optimize/optimize.py#L454. 

Consider adding the https://github.com/scipy/scipy/blob/maintenance/1.6.x/LICENSE.txt to the top of this file referring to wrap_function.

**Comment by mpharrigan** [comment]
added

File: recirq/optimize/__init__.py [hunk 1]
```diff
@@ -12,4 +12,53 @@
...
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+def wrap_function(function, args):
```
Comment (dstrain115) [inline recirq/optimize/__init__.py:49]: I generally don't think it's a great idea to put code in __init__ files.  Is there a reason we can't put this in its own file and import it into the package?

File: recirq/optimize/__init__.py [hunk 2]
```diff
@@ -12,4 +12,53 @@
...
+    # This function existed in scipy prior to 1.7
+    # It must not have been considered part of the public API, as it has
+    # disappeared without a trace.
```
Comment (dstrain115) [inline recirq/optimize/__init__.py:52]: Should this be a docstring instead of line comments?

**Review by dstrain115 — APPROVED** [review]

File: recirq/optimize/__init__.py (hunk 2 above)
Comment (mpharrigan) [reply recirq/optimize/__init__.py]: done

File: recirq/optimize/__init__.py (hunk 1 above)
Comment (mpharrigan) [reply recirq/optimize/__init__.py]: done, moved

**Review by balopat — APPROVED** [review]

**Pull request merged**